mdbook-course/src/course.rs

```diff
@@ -313,24 +307,18 @@ impl Session {
 
     /// Generate a Markdown outline for this session, for placement at the given
     /// path.
-    pub fn outline(&self, at_source_path: impl AsRef<Path>) -> String {
-        let mut outline = String::from("In this session:\n");
+    pub fn outline(&self) -> String {
+        let mut segments = Table::new(["Segment".into(), "Time".into()]);
         for segment in self {
             // Skip short segments (welcomes, wrap-up, etc.)
             if segment.minutes() == 0 {
                 continue;
             }
-            writeln!(
-                &mut outline,
-                " * [{}]({}) ({})",
-                segment.name,
-                relative_link(&at_source_path, &segment.slides[0].source_paths[0]),
-                duration(segment.minutes())
-            )
-            .unwrap();
+            segments.add_row([segment.name.clone(), duration(segment.minutes())]);
         }
-        writeln!(&mut outline,"\nIncluding {BREAK_DURATION} minute breaks, this session should take about {}", duration(self.minutes())).unwrap();
-        outline
+        format!(
+            "Including {BREAK_DURATION} minute breaks, this session should take about {}. It contains:\n\n{}",
+            duration(self.minutes()),segments)
```

Comment:
I think your `rustfmt` has given up here :smile: 

```suggestion
            duration(self.minutes()), segments)
```

Reply:
I think it doesn't do much in macro invocations? I was surprised by this too.